app.ts

```diff
@@ -0,0 +1,286 @@
+let fromParameter = 0;
+let recordCount: number;
+const el = document.getElementById
+
+// Gets The 2nd paramater from the 1st parameter
+function getParameters(fromParameter: number): number {
+	let toParameter: number;
+	let noOfRows = getNoOfRows();
+
+	toParameter = fromParameter + noOfRows;
+	return toParameter;
+}
+
+// Gets the number of rows on the screen
+function getNoOfRows(): number {
+	const height = window.innerHeight;
+	let noOfRows = Math.floor(height / 40);
+	return noOfRows;
+}
+
+//// Functions To Create The HTML
+// Heading Row
+function createHeadingColumn(headingData: string) {
+	const heading: HTMLElement | null = document.getElementById("heading");
+
+	let headings = `<div class="headings" id="headings">${headingData}</div>`;
+	if (heading !== null) {
+		heading.innerHTML += headings;
+	}
+}
+
+// Table Content
+function appendTableContent(contentData: string | string[]) {
+	const content: HTMLElement | null = document.getElementById("content");
+
+	let table = `<div id="row-${contentData[0]}" class="rows"></div>`;
+	if (content !== null) {
+		content.innerHTML += table;
+	} else {
+		return alert("ERROR");
+	}
+
+	let row: HTMLElement | null = document.getElementById("row-" + contentData[0]);
+	for (let x of contentData) {
+		let rowCols = `<div class="row_cols">${x}</div>`;
+		if (row !== null) {
+			row.innerHTML += rowCols;
+		}
+	}
+}
+
+//// Fetch Requests
+// Response Error Handling
+function handleErrors(response: Response) {
+	if (!response.ok) {
+		throw Error(response.statusText);
+	}
+	return response;
+}
+
+// Heading Row (Getting the columns data)
+function getHeadings(): Promise<void> {
+	return fetch("http://localhost:2050/columns", {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((response: Response) => response.json())
+		.then((headingData: string | string[]) => {
+			for (let heading of headingData) {
+				createHeadingColumn(heading);
+			}
+		})
+		.catch((error: Error) => {
+			console.log(error);
+		});
+}
+
+// Table Content (Getting the table's data)
+function getTable(): Promise<void> {
+	const content: HTMLElement | null = document.getElementById("content");
+	let toParameter = getParameters(fromParameter);
+	const pageStats: HTMLElement | null = document.getElementById("pageStats");
+
+	// Clears Table
+	if (content !== null) {
+		content.innerHTML = "";
+	}
+
+	// Displays The Current Results Being Shown
+	let currentStats = `Showing results from ${fromParameter} to ${toParameter} out of ${recordCount} results.`;
+	if (pageStats !== null) {
+		pageStats.innerHTML = currentStats;
+	}
+
+	return fetch(`http://localhost:2050/records?from=${fromParameter}&to=${toParameter}`, {
+		method: "GET",
+		headers: { "Content-Type": "application/json" },
+	})
+		.then(handleErrors)
+		.then((res: Response) => res.json())
+		.then((contentData: string | string[]) => {
```

Comment:
They way your code is written... neither of these are right. Based off what you get from the back end, should it not be a 2D array?